ext/datadog_profiling_native_extension/heap_recorder.c

```diff
@@ -348,11 +355,16 @@ void end_heap_allocation_recording(struct heap_recorder *heap_recorder, ddog_pro
   commit_recording(heap_recorder, heap_record, active_recording);
 }
 
-void heap_recorder_prepare_iteration(heap_recorder *heap_recorder) {
+size_t heap_recorder_prepare_iteration(heap_recorder *heap_recorder, size_t min_age) {
+  size_t current_gc_gen = rb_gc_count();
+
   if (heap_recorder == NULL) {
-    return;
+    return current_gc_gen;
   }
 
+  heap_recorder->iteration_gen = current_gc_gen;
+  heap_recorder->iteration_min_gen_age = min_age;
+
   if (heap_recorder->object_records_snapshot != NULL) {
     // we could trivially handle this but we raise to highlight and catch unexpected usages.
     rb_raise(rb_eRuntimeError, "New heap recorder iteration prepared without the previous one having been finished.");
```

Comment:
:arrow_down: Actually, if we're not going to include objects with age == 0, I think it would be an overall win to additionally tweak `st_object_record_update`, below, to also be aware of this.

E.g. why bother going `ruby_ref_from_id` + `ruby_obj_memsize_of` if we're not going to use the data anyway?

Reply:
🤦 that was precisely my intention and why I marked PROF-9471 as a duplicate of PROF-9469. But I added the age check to the wrong place 😂